docs/release-info/release-notes.md

```diff
@@ -57,50 +57,741 @@ For tips about how to write a good release note, see [Release notes](https://git
+
+#### Improved Azure input source
+
+You can now ingest data from multiple storage accounts using the new `azureStorage` input source schema instead of the now deprecated `azure` input source schema. For example:
+
+```json
+...
+    "ioConfig": {
+      "type": "index_parallel",
+      "inputSource": {
+        "type": "azureStorage",
+        "objectGlob": "**.json",
+        "uris": ["azureStorage://storageAccount/container/prefix1/file.json", "azureStorage://storageAccount/container/prefix2/file2.json"]
+      },
+      "inputFormat": {
+        "type": "json"
+      },
+      ...
+    },
+...
+```
 
-#### SQL-based ingestion
+[#15630](https://github.com/apache/druid/pull/15630)
 
-##### Other SQL-based ingestion improvements
+#### Data management API improvements
 
-#### Streaming ingestion
+Improved the [Data management API](https://druid.apache.org/docs/latest/api-reference/data-management-api) as follows:
 
-##### Other streaming ingestion improvements
+* Fixed a bug in the `markUsed` and `markUnused` APIs where an empty set of segment IDs would be inconsistently treated as null or non-null in different scenarios [#16145](https://github.com/apache/druid/pull/16145)
+* Improved the `markUnused` API endpoint to handle an empty list of segment versions [#16198](https://github.com/apache/druid/pull/16198)
+* The `segmentIds` filter in the Data management API payload is now parameterized in the database query [#16174](https://github.com/apache/druid/pull/16174)
+* You can now mark segments as used or unused within the specified interval using an optional list of versions.
+For example: `(interval, [versions])`. When `versions` is unspecified, all versions of segments in the `interval` are marked as used or unused, preserving the old behavior [#16141](https://github.com/apache/druid/pull/16141)
+
+#### Nested columns performance improvement
+
+Nested column serialization now releases nested field compression buffers as soon as the nested field serialization is complete, which requires significantly less direct memory during segment serialization when many nested fields are present.
+
+[#16076](https://github.com/apache/druid/pull/16076)
+
+#### Segment allocation
+
+Druid now associates pending segments with the task groups that created them.
+
+Associating pending segments with the task groups facilitates clean up of unneeded segments as soon as all tasks in the group exit.
+Cleaning up pending segments helps delete entries immediately after tasks exit and can alleviate the load on the metadata store during segment allocation.
+This can also help with segment allocation failures due to conflicting pending segments that are no longer needed in some cases.
+
+The change ensures that an append action upgrades a segment set which corresponds exactly to the pending segment upgrades made by the concurrent replace action, and eliminates any duplication in query results that may occur.
+
+[#16144](https://github.com/apache/druid/pull/16144)
```

Comment:
This entire section can go inside `concurrent append and replace improvements` as that is the feature which really benefits from this.

Can be rephrased as below:

```
Druid now performs active cleanup of stale pending segments by tracking the set of tasks using such pending segments.
This allows concurrent append and replace to upgrade only a minimal set of pending segments and thus improve performance and eliminate errors.
Additionally, it helps in reducing load on the metadata store.

[#16144](https://github.com/apache/druid/pull/16144)
```

Reply:
Updated.